basic_module/distribution/src/core.rs

```diff
@@ -13,3 +13,18 @@
 //
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use crate::error::Error;
+use ckey::Ed25519Public as Public;
+use ctypes::BlockHash;
+use std::collections::BTreeMap;
+
+pub trait OpenBlockHandler {
+    fn distribute_when_open_block(
```

Comment:
You should consult the abci interface functions. The function signature don't seem like this.